File: app-rn/src/theme.ts
import { Platform, type ViewStyle } from 'react-native';
import { tokens } from './generated/tokens';

export const C = tokens.color;
export const R = tokens.radius;
export const S = tokens.space;
export const T = tokens.size;
export const MIN_TOUCH = tokens.touch.minTarget;

/** Font family names as registered in App.tsx via useFonts. */
export const F = {
  medium: 'BalooThambi2_500Medium',
  semibold: 'BalooThambi2_600SemiBold',
  bold: 'BalooThambi2_700Bold',
  heavy: 'BalooThambi2_800ExtraBold',
} as const;

/** The flat "drop" under cards and buttons (no blur, per the style rules). */
export const drop = (color: string, height: number = tokens.shadow.cardDrop) => ({
  shadowColor: color,
  shadowOffset: { width: 0, height },
  shadowOpacity: 1,
  shadowRadius: 0,
  elevation: 0,
  borderBottomWidth: 0,
});

/** Web only: stop drags (tracing, sliding, scratching) from selecting text. */
export const NO_SELECT: ViewStyle = Platform.OS === 'web' ? ({ userSelect: 'none' } as unknown as ViewStyle) : {};

File: ios/TamilTrain/Theme.swift
import CoreText
import SwiftUI

/// Baloo Thambi 2 covers Tamil and Latin in one rounded family.
/// Fonts are registered at launch, so no Info.plist UIAppFonts entry is needed.
enum FontRegistry {
    static let files = ["BalooThambi2_500Medium", "BalooThambi2_600SemiBold", "BalooThambi2_700Bold", "BalooThambi2_800ExtraBold"]

    static func registerBundledFonts() {
        for name in files {
            guard let url = Bundle.main.url(forResource: name, withExtension: "ttf") else {
                assertionFailure("Missing font \(name).ttf in app bundle")
                continue
            }
            CTFontManagerRegisterFontsForURL(url as CFURL, .process, nil)
        }
    }
}

enum Baloo: String {
    case medium = "BalooThambi2-Medium"
    case semibold = "BalooThambi2-SemiBold"
    case bold = "BalooThambi2-Bold"
    case heavy = "BalooThambi2-ExtraBold"
}

extension Font {
    /// Scales with Dynamic Type relative to `style`.
    static func baloo(_ size: CGFloat, _ weight: Baloo = .heavy, relativeTo style: Font.TextStyle = .body) -> Font {
        .custom(weight.rawValue, size: size, relativeTo: style)
    }
}

extension View {
    /// The flat "drop" under cards and buttons. No blur, per the illustration rules.
    func drop(_ color: Color = Tokens.Palette.cardShadow, _ height: CGFloat = 6) -> some View {
        shadow(color: color, radius: 0, x: 0, y: height)
    }

    func card(radius: CGFloat = Tokens.Radius.lg) -> some View {
        background(Color.white, in: RoundedRectangle(cornerRadius: radius, style: .continuous))
            .drop()
    }
}
